titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/common/V3QueueableTask.java

```diff
@@ -226,19 +235,27 @@ public void opportunisticSchedulingFailed() {
         if (!runtimePrediction.isPresent()) {
             return; // noop, opportunisticCpuCount is always 0
         }
-        opportunisticCpuCount.updateAndGet(current -> current >= 1 ? current - 1 : initialOpportunisticCpuCount(cpus));
+        if (!opportunisticEnabledSupplier.get()) {
+            opportunisticCpuEnabled = false;
```

Comment:
I do not see a place where it is reset. Unless we do not want to do it for a particular task?

Reply:
good catch

Reply:
Fixed